cloudmarker/events/azlogprofileretentionevent.py

```diff
@@ -0,0 +1,116 @@
+"""Microsoft Azure Log Profile Retention Event.
+
+This module defines the :class:`AzLogProfileRetentionEvent` class that
+identifies if an Azure log profile's retention policy is configured for
+less than the minimum number of days than required. This plugin works
+properties found in the ``ext`` bucket of ``log_profile`` records.
+"""
+
+
+import logging
+
+from cloudmarker import util
+
+_log = logging.getLogger(__name__)
+
+
+class AzLogProfileRetentionEvent:
+    """Azure log profile retention event plugin."""
+
+    def __init__(self, _min_retention_days=365):
+        """Create an instance of :class:`AzLogProfileRetentionEvent`.
+
+        Arguments:
+            _min_retention_days (int): Minimum required
+                                    retention days.
+
+        """
+        self._min_retention_days = _min_retention_days
+        _log.info("Initialized; minimum retention days: %d",
+                  self._min_retention_days)
+
+    def eval(self, record):
+        """Evaluate Azure log profiles for retention policy.
+
+        Arguments:
+            record (dict): An Azure log profile record.
+
+        Yields:
+            dict: An event record representing an Azure log profile
+            with retention set to less than the required days.
+
+        """
+        com = record.get('com', {})
+        if com is None:
+            return
+
+        if com.get('cloud_type') != 'azure':
+            return
+
+        if com.get('record_type') != 'log_profile':
+            return
+
+        ext = record.get('ext', {})
+        if ext is None:
+            return
+
+        if ext['retention_enabled']:
+            if ext['retention_days'] <\
+                    self._min_retention_days:
+                yield from \
+                    _get_log_profile_retention_event(com, ext,
+                                                     self._min_retention_days)
+        else:
+            if ext['retention_days'] == 0:
+                yield from \
+                    _get_log_profile_retention_event(com, ext,
+                                                     self._min_retention_days)
```

Comment:
I am confused about the `else` part. If we have reached the `else` part, it means that `ext['retention_enabled']` is `False`. In that case, should we not generate an event regardless of what the actual number of retention days is?

Reply:
To store logs for forever `retention_enabled` must be `False` and `retention_days` must be 0.
The `else` here takes care of that scenario, if the `retention_enabled` is `False` and `retention_days` is not `0` then an event is generated.